src/Razor/src/Microsoft.VisualStudio.RazorExtension/Snippets/SnippetService.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.Editor.Razor.Snippets;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.RazorExtension.Snippets;
+
+internal class SnippetService
```

Comment:
So `SnippetService` is used once to populate `SnippetCache`, other things access `SnippetCache` and then call `XmlSnippetParser` which has its own cache for the results?

Can some or part of that be combined so that there is a service that is used to retreive parsed snippets, and all of the  population and caching etc. is an impl detail of that? Or maybe this can just be renamed to `SnippetLoaderService`, and it can be called by `XmlSnippetParser` on first use, and then `SnippetCache` can be internal to `XmlSnipperParser`?

I'm just wondering if I'll remember how any of this works in 6 months 😁

Reply:
I'm all in favor of simplifying this structure a bit. Let me think more on it, especially now that I've simplified a lot of what we're doing.

Reply:
See what you think now. Parser is static, SnippetInfo lazily parses as needed